### ISSUE TITLE
`$$.filterBy(and(..))` produces confusing collection description in exception if filtered collection is empty

### ISSUE THREAD

**Issue by fokinp** [selenide/selenide#1392]
**Steps to reproduce:**
The following code can be a part of `CollectionMethodsTest`
```java
ElementsCollection filteredRows = $$("#multirowTable tr")
      .filterBy(and("My awesome name", text("Chack"), text("Baskerville")));

assertThatThrownBy(() -> filteredRows.shouldHave(size(0)))
      .hasMessageStartingWith(<message>)
```
**Actual Result:**
message starts with:
`List size mismatch: expected: > 0, actual: 0, collection: #multirowTable tr.filter(text 'Boseman')`
There are no any references that we use `and` condition here.

**Expected Result:**
message starts with:
`List size mismatch: expected: > 0, actual: 0, collection: #multirowTable tr.filter(My awesome name: text 'Chack' and text 'Baskerville')`

So basically, when we deal with filtering logic, collection description should contain full `And` condition description nevertheless on first `failed` condition for the last element in collection.

The easiest way to fix it is to change `And.toString` logic and  remove "dynamic" nature of this (always return "full" description). However this change affects exceptions readability (doesn't it?) in case we use `And` condition for `SelenideElement`

**Comment by asolntsev** [comment]
I was also thinking that the "dynamic" nature of `And.toString()` is scary. It may cause some bugs or unexpected behaviour. 
I agree with your suggestion to remove the "dynamic" nature.

**Comment by fokinp** [comment]
I will prepare a fix in a day

**Comment by asolntsev** [comment]
fixed by https://github.com/selenide/selenide/pull/1393
